src/Spice86.Core/Emulator/InterruptHandlers/Common/MemoryWriter/MemoryWriter.cs

```diff
@@ -38,7 +38,7 @@ public class MemoryWriter {
     /// <param name="b">data to write</param>
     public void WriteUInt8(byte b) {
         _memory.UInt8[CurrentAddress.Segment, CurrentAddress.Offset] = b;
-        CurrentAddress.Offset++;
+        CurrentAddress = new SegmentedAddress(CurrentAddress.Segment, (ushort)(CurrentAddress.Offset + 1));
```

Comment:
since operator + is redefined for SegmentedAddress object and since it adds a value to the offset, isnt it possible to do CurrentAddress++?

Reply:
Operator '++' cannot be applied to operand of type 'SegmentedAddress'

However, it works with CurrentAddress += 1